packages/jest-mock/__typetests__/mock-functions.test.ts

```diff
@@ -250,7 +250,12 @@ expectType<Mock<() => Promise<string>>>(
 );
 expectError(fn(() => Promise.resolve('')).mockRejectedValueOnce());
 
-// jest.spyOn()
+expectType<void>(mockFn.withImplementation(mockFnImpl, () => {}));
+expectType<Promise<void>>(
+  mockFn.withImplementation(mockFnImpl, async () => {}),
+);
+
+expectError(mockFn.withImplementation(mockFnImpl, () => {}).then());
```

Comment:
Just above you show that it returns `void`, so maybe it is better to check that it requires two args?

Also leave the comment, please. I need it (;

```suggestion
expectError(mockFn.withImplementation(mockFnImpl);

// jest.spyOn()
```

Reply:
Oh, my mistake with the comment.

I also felt the `expectError`-part was a bit redundant, I'm happy to get rid of it.